invisible_cities/cities/isidora.py

```diff
@@ -0,0 +1,231 @@
+        self.xs              = DataSiPM.X.values
+        self.ys              = DataSiPM.Y.values
+        self.adc_to_pes      = abs(DataPMT.adc_to_pes.values).astype(np.double)
+        self.sipm_adc_to_pes = DataSiPM.adc_to_pes.values    .astype(np.double)
+        self.coeff_c         = DataPMT.coeff_c.values        .astype(np.double)
+        self.coeff_blr       = DataPMT.coeff_blr.values      .astype(np.double)
+        self.xs              = DataSiPM.X.values
+        self.ys              = DataSiPM.Y.values
+
+        # BLR default values (override with set_BLR)
+        self.n_baseline   = 28000
+        self.thr_trigger  = 5 * units.adc
+
+        # MAU default values (override with set_MAU)
+        self.n_MAU        = 100
+        self.thr_MAU      = thr_MAU = 3 * units.adc
+
+        # set switches
+        self.setFiles     = False
+        self.setSiPM      = False
+        self.setCwfStore  = False
+
+        # defaul parameters
+        self.nprint       = 1000000
+        self.signal_start =       0
+        self.signal_end   =    1200
+
+    def set_print(self, nprint=10):
+        """Print frequency."""
+        self.nprint = nprint
+
+    def set_input_files(self, input_files):
+        """Set the input files."""
+        self.input_files = input_files
+        self.setFiles    = True
+
+    def set_BLR(self, n_baseline=38000, thr_trigger=5 * units.adc):
+        """Parameters of the BLR."""
+        self.n_baseline  = n_baseline
+        self.thr_trigger = thr_trigger
+
+    def set_MAU(self, n_MAU=100, thr_MAU=3 * units.adc):
+        """Parameters of the MAU used to remove low frequency noise."""
+        self.  n_MAU =   n_MAU
+        self.thr_MAU = thr_MAU
+
+    def set_cwf_store(self, cwf_file, compression='ZLIB4'):
+        """Set the input files."""
+        # open cwf store
+        self.cwfFile = tb.open_file(
+            cwf_file, "w", filters=tbl.filters(compression))
+
+        # create a group
+        self.cwf_group = self.cwfFile.create_group(self.cwfFile.root, "BLR")
+
+        self.setCwfStore = True
+        self.compression = compression
+
+    def store_cwf(self, cwf):
+        """Store CWF."""
+        NPMT, PMTWL = cwf.shape
+        if "pmtcwf" not in self.cwf_group:
+            # create earray to store cwf
+            self.pmtcwf = self.cwfFile.create_earray(self.cwf_group, "pmtcwf",
+                                                atom=tb.Float32Atom(),
+                                                shape=(0, NPMT, PMTWL),
+                                                filters=tbl.filters(self.compression))
+        self.pmtcwf.append(cwf.reshape(1, NPMT, PMTWL))
+
+
+    def run(self, nmax):
+        """
+        Run the machine
+        nmax is the max number of events to run
+        """
+        n_events_tot = 0
+        # check the state of the machine
+        if not self.input_files:
+            raise IOError('input file list is empty')
```

Comment:
I see that this habit of using `IOError` in these situations is spreading throughout IC. I don't think that that is appropriate in such caes. Consider that from Python 3.3 onwards, `IOError` is just a synonym for `OSError`, and I hope that it's pretty clear that this is not an OS error! I think it would be better if we had specific IC exceptions with the exact meanings that we want to convey, and that we use them in these situations.

Let's nip this in the bud before too many people copy this bad practice!